qa/tasks/mgr/dashboard/test_mgr_module.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import logging
+import requests
+import time
+
+from .helper import DashboardTestCase, JObj, JList, JLeaf
+
+logger = logging.getLogger(__name__)
+
+
+class MgrModuleTestCase(DashboardTestCase):
+    @classmethod
+    def tearDownClass(cls):
+        cls._ceph_cmd(['mgr', 'module', 'disable', 'telemetry'])
+        super(MgrModuleTestCase, cls).tearDownClass()
+
+    def wait_until_rest_api_accessible(self):
+        """
+        Wait until the REST API is accessible.
+        """
+
+        def _check_connection():
+            try:
+                # Try to reaching an API endpoint successfully.
+                self._get('/api/mgr/module')
+                if self._resp.status_code == 200:
+                    return True
+            except requests.ConnectionError:
+                pass
+            return False
+
+        self.wait_until_true(lambda: _check_connection(), timeout=20, period=2)
```

Comment:
Can't `_check_connection` just be passed to `wait_until_true` without having to use the lambda expression?

Reply:
It's required, otherwise this error occurs:
```
2019-01-22 10:13:54,652.652 INFO:__main__:Traceback (most recent call last):
2019-01-22 10:13:54,652.652 INFO:__main__:  File "/ceph/qa/tasks/mgr/dashboard/test_mgr_module.py", line 142, in test_disable
2019-01-22 10:13:54,653.653 INFO:__main__:    self.wait_until_rest_api_accessible()
2019-01-22 10:13:54,653.653 INFO:__main__:  File "/ceph/qa/tasks/mgr/dashboard/test_mgr_module.py", line 34, in wait_until_rest_api_accessible
2019-01-22 10:13:54,653.653 INFO:__main__:    self.wait_until_true(_check_connection(), timeout=20, period=2)
2019-01-22 10:13:54,653.653 INFO:__main__:  File "/ceph/qa/tasks/ceph_test_case.py", line 159, in wait_until_true
2019-01-22 10:13:54,653.653 INFO:__main__:    if condition():
2019-01-22 10:13:54,653.653 INFO:__main__:TypeError: 'bool' object is not callable
```

Reply:
This reads as if you've used
`self.wait_until_true(_check_connection(), timeout=20, period=2)`
instead of 
`self.wait_until_true(_check_connection, timeout=20, period=2)`

Reply:
Haha, you're right :-) Switch on brain ... done ...